### PR TITLE
[Master] Fix goto definition feature on nested types

### LINKED ISSUE #40154: [Bug]: Goto definition not working for `string:RegExp`
### Description

$subject

### Steps to Reproduce

_No response_

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Fixes #40154

https://github.com/ballerina-platform/ballerina-lang/assets/46857198/903292bc-85b9-4f5a-ae7b-a5cee4b52d70

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java [hunk 1]
```diff
@@ -25,9 +25,9 @@ public class ProgramFileConstants {
 
     public static final int MAGIC_NUMBER = 0xBA1DA4CE;
     public static final short VERSION_NUMBER = 50;
-    public static final int BIR_VERSION_NUMBER = 68;
-    public static final short MIN_SUPPORTED_VERSION = 68;
-    public static final short MAX_SUPPORTED_VERSION = 68;
+    public static final int BIR_VERSION_NUMBER = 69;
```
Comment (pcnfernando) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java:28]: We only need to update the BIR version, if the actual BIR was changed. In other words, reading/writing. Here, with your change, you only have used an already available value to populate another reference. Hence, the actual BIR didn't change

Comment (LakshanWeerasinghe) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java:28]: Got it.

**Comment by sanjana** [comment]
@LakshanWeerasinghe Was the `.references()` also not working in the nested type defs?
If so, need to add a couple of tests in the Semantic API as well.

File: language-server/modules/langserver-core/src/test/resources/definition/sources/projectls/modules/lsmod3/lsmod3.bal [hunk 2]
```diff
@@ -1,4 +1,7 @@
 import ballerina/module1;
+import projectls.lsmod1;
 
 public function mod3Function1() {
 }
+
+public type MyType lsmod1:MyType;
```
Comment (SasinduDilshara) [inline language-server/modules/langserver-core/src/test/resources/definition/sources/projectls/modules/lsmod3/lsmod3.bal:7]: Lets add one test scenario like this

```
public type MyType2 MyType;
```

and add 

```
lsmod3:MyType 2_ = 123;
```

inside the `projectls/typereference`.

this is to check nested types on the safe file/module.
wdyt?

File: language-server/modules/langserver-core/src/test/resources/definition/sources/projectls/typereference.bal [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import projectls.lsmod3;
+import projectls.lsmod1;
```
Comment (SasinduDilshara) [inline language-server/modules/langserver-core/src/test/resources/definition/sources/projectls/typereference.bal:2]: Its better to change the import order

**Review by SasinduDilshara — APPROVED** [review]

**Pull request merged**